services/rfq/relayer/reldb/base/quote.go

```diff
@@ -89,7 +93,19 @@ func (s Store) GetQuoteResultsByStatus(ctx context.Context, matchStatuses ...rel
 }
 
 // UpdateQuoteRequestStatus todo: db test.
-func (s Store) UpdateQuoteRequestStatus(ctx context.Context, id [32]byte, status reldb.QuoteRequestStatus) error {
+func (s Store) UpdateQuoteRequestStatus(ctx context.Context, id [32]byte, status reldb.QuoteRequestStatus, prevStatus *reldb.QuoteRequestStatus) error {
+	if prevStatus == nil {
+		req, err := s.GetQuoteRequestByID(ctx, id)
+		if err != nil {
+			return fmt.Errorf("could not get quote: %w", err)
+		}
+		prevStatus = &req.Status
+	}
+	if !isValidStateTransition(*prevStatus, status) {
+		logger.Warnf("invalid state transition from %s to %s [txid=%s]", *prevStatus, status, hexutil.Encode(id[:]))
```

Comment:
Chose not to return error here since we presumably don't want to panic on this, and checking `errors.Is()` on the result introduces a bit of code bloat

Reply:
Completely understand the hesitation to do this, but in this case I'll be renaming the method to `UpdateQuoteRequestStatusIFValid` or somethign.